File: app/src/main/java/ie/dublinbuspal/android/data/local/dao/RouteServiceDao.java
package ie.dublinbuspal.android.data.local.dao;

import androidx.room.Dao;
import androidx.room.Query;
import ie.dublinbuspal.android.data.local.DbInfo;
import ie.dublinbuspal.android.data.local.entity.RouteService;

@Dao
public interface RouteServiceDao extends BaseDao<RouteService> {

    @Query(DbInfo.SELECT_ROUTE_SERVICE)
    RouteService select(String routeId);

    @Query(DbInfo.DELETE_ROUTE_SERVICES)
    void deleteAll();

}

File: data/src/main/kotlin/ie/dublinbuspal/data/dao/RouteServiceDao.kt
package ie.dublinbuspal.data.dao

import androidx.room.Dao
import androidx.room.Query
import ie.dublinbuspal.data.entity.RouteServiceEntity
import io.reactivex.Maybe

@Dao
interface RouteServiceDao : BaseDao<RouteServiceEntity> {

    @Query("SELECT * FROM route_services WHERE id = :id")
    fun select(id: String): Maybe<RouteServiceEntity>

    @Query("DELETE FROM route_services")
    fun deleteAll()

}
